cpp/daal/src/algorithms/objective_function/cross_entropy_loss/cross_entropy_loss_dense_default_batch_impl.i

```diff
@@ -576,16 +589,14 @@ services::Status CrossEntropyLossKernel<algorithmFPType, method, cpu>::compute(N
         TArrayScalable<algorithmFPType, cpu> aX(n * p);
         TArrayScalable<algorithmFPType, cpu> aY(n);
         s |= objective_function::internal::getXY<algorithmFPType, cpu>(dataNT, dependentVariablesNT, ntInd, aX.get(), aY.get(), nRows, n, p);
-        s |= doCompute(aX.get(), aY.get(), nRows, n, p, betaNT, valueNT, hessianNT, gradientNT, nonSmoothTermValue, proximalProjection,
-                       lipschitzConstant, parameter);
+        auto internalDataNT               = HomogenNumericTable<algorithmFPType>::create(aX.get(), p, n);
```

Comment:
Considering that it will work with a solver, and at each iteration, do you suggest allocating data? I would suggest cashing a block of allocated memory.